app/src/main/java/com/example/currencyconverter/ui/di/ServiceLocator.kt

```diff
@@ -14,18 +17,37 @@ object ServiceLocator {
         NetworkService()
     }
 
+    private val domainMapper by lazy {
+        CurrencyDomainMapper()
+    }
+
+    private val networkMapper by lazy {
+        CurrencyNetworkMapper()
+    }
+
+    private val databaseMapper by lazy {
+        CurrencyDatabaseMapper()
+    }
+
+
+    private var valueInputValidator : ValueInputValidator? = null
+
     @Volatile
```

Comment:
Please check @Volatile use cases. [Here](https://stackoverflow.com/questions/106591/what-is-the-volatile-keyword-useful-for), for example